### PR TITLE
Weight calcs for words. Enter triggers onclick. fav.ico.

### PR DESCRIPTION
**Pull request by nudabagana**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -145,8 +143,70 @@ const setSuggestion = (result) => {
...
-filterBtn.onclick = () => {
+const onClick = () => {
   clearAll("suggestion");
   generateWord();
 };
+
+const filterBtn = document.getElementById("filter-btn");
+filterBtn.onclick = onClick;
+
+const ENTER_CODE = "Enter";
```
Comment (MantasSidabras) [inline index.js]: No need for a constant if it's only used in one place

File: index.js [hunk 2]
```diff
@@ -116,13 +112,15 @@ const generateWord = () =>
...
 
       console.log({ excludeSet, includeSet, excludes, includes, regexes });
-      const result = dict
-        .filter(removeDupicateLetterWords(removeDuplicates))
-        .filter(filterSet([...excludeSet, ...excludes])(false))
-        .filter(filterSet([...includeSet, ...includes])(true))
-        .filter((x) => regexes.every((pattern) => pattern.test(x)));
-
-      setSuggestion(result.random());
+      const result = addWeights(
```
Comment (MantasSidabras) [inline index.js]: addWeights does multiple things and the name does not reflect it. Also it breaks the dot chain pipeline Preferably make it as a map function like .map(weighWords)

File: index.js [hunk 3]
```diff
@@ -116,13 +112,15 @@ const generateWord = () =>
...
+        dict
+          .filter(removeDupicateLetterWords(removeDuplicates))
+          .filter(filterSet([...excludeSet, ...excludes])(false))
+          .filter(filterSet([...includeSet, ...includes])(true))
+          .filter((x) => regexes.every((pattern) => pattern.test(x)))
+      ).map(({ word }) => word);
+
+      setSuggestion(result[0]);
```
Comment (MantasSidabras) [inline index.js]: Would be nice to avoid array out of bound

File: index.js [hunk 4]
```diff
@@ -145,8 +143,70 @@ const setSuggestion = (result) => {
...
+  if (e.key == ENTER_CODE) {
+    onClick();
+  }
+});
+
+const COUNT_WEIGHT = 0.5;
+const POS_WEIGHT = 1;
+const NO_DUPLICATE_CHAR = true;
+
+const addWeights = (words) => {
```
Comment (MantasSidabras) [inline index.js]: Break it down to smaller functions

File: index.js [hunk 5]
```diff
@@ -145,8 +143,70 @@ const setSuggestion = (result) => {
...
+
+document.addEventListener("keydown", (e) => {
+  if (e.key == ENTER_CODE) {
+    onClick();
+  }
+});
+
+const COUNT_WEIGHT = 0.5;
+const POS_WEIGHT = 1;
+const NO_DUPLICATE_CHAR = true;
```
Comment (MantasSidabras) [inline index.js]: How it's different from the checkbox value?

Comment (nudabagana) [reply index.js]: does not add weight for same char in words. Like "saab" => 5 + 10 +0 +15 (ignores 2nd "a" weight). Renamed to "SKIP_DUPLICATE_CHAR_WEIGHT".

**Pull request merged**